test/fixtures.rb

```diff
@@ -0,0 +1,121 @@
+require 'lotus/view'
+require 'lotus/helpers/html_helper'
+
+class HtmlView
+  include Lotus::Helpers::HtmlHelper
+
+  def empty_div
+    html.div
+  end
+
+  def div_with_string_content
+    html.div('hello world')
+  end
+
+  def div_with_block_content_as_string
+    html.div { 'hola' }
+  end
+
+  def div_with_block_content_with_tag_helper
+    html.div(html.p('inner'))
+  end
+
+  def div_with_block_content_with_nested_calls
+    html.div do
+      span 'hello'
+    end
+  end
+
+  def div_with_block_content_and_multiple_nested_calls
+    html.form(action: '/users', method: 'POST') do
+      div do
+        label 'First name', for: 'user-first-name'
+        input type: 'text', id: 'user-first-name', name: 'user[first_name]', value: 'L'
+      end
+
+      input type: 'submit', value: 'Save changes'
+    end
+  end
+
+  def div_with_attr
+    html.div(id: 'container')
+  end
+
+  def div_with_undescore_attr
+    html.div('data_where' => 'up')
+  end
+
+  def div_with_attrs
+    html.div(id: 'content', class: 'filled')
+  end
+
+  # FIXME add more cases from:
```

Comment:
Well, that was a note from the initial implementation, before we introduced `Utils::Escape`. Specifically, that is about escaping tag attributes.

We don't do it automatically, because it's technically impossible, but we have `Utils::Escape.html_attribute` which will be available soon as `#ha` helper.

The technical limitation mentioned above is about the impossibility to distinguish between what should be escaped and what not. Consider the following example:

```ruby
def home_page_link
  html.a('home', href: '/')
end

home_page_link # => <a href="&#x2f;">home</a> # WRONG
```

Developers should escape attributes by themselves only if those contain user inputs.